File: day_1/day_1.rb
p File.readlines('calories.txt', chomp: true)
      .slice_after { |line| line.strip.empty? }
      .map { |group| group.map(&:to_i) }
      .map(&:sum)
      .max

p File.readlines('calories.txt', chomp: true)
      .slice_after { |line| line.strip.empty? }
      .map { |group| group.map(&:to_i) }
      .map(&:sum)
      .sort[-3..]
      .sum

File: day_1/day_1.exs
defmodule Day1 do
  def part_1 do
    File.read!('calories.txt')
    |> String.split("\n\n", trim: true)
    |> Enum.map(fn group -> group |> String.split("\n", trim: true) end)
    |> Enum.map(fn item -> item |> Enum.map(&String.to_integer/1) end)
    |> Enum.map(&Enum.sum/1)
    |> Enum.max()
  end

  def part_2 do
    File.read!('calories.txt')
    |> String.split("\n\n", trim: true)
    |> Enum.map(fn group -> group |> String.split("\n", trim: true) end)
    |> Enum.map(fn item -> item |> Enum.map(&String.to_integer/1) end)
    |> Enum.map(&Enum.sum/1)
    |> Enum.sort()
    |> Enum.slice(-3, 3)
    |> Enum.sum()
  end
end

Day1.part_1()
|> IO.puts()

Day1.part_2()
|> IO.puts()
